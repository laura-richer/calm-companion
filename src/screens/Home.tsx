import React from 'react';
import { View } from 'react-native';

import { NavigationList } from '~components/Navigation/NavigationList';

const Home = () => {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <NavigationList />
    </View>
  );
};

export default Home;
