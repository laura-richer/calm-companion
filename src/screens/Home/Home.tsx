import React from 'react';
import { View } from 'react-native';
import ScreenHeader from '~components/ScreenHeader/ScreenHeader';
import ContentWrapper from '~components/ContentWrapper/ContentWrapper';
import CardNavigationContainer from '~components/Navigation/NavigationContainers/CardNavigationContainer';

const Home = () => {
  return (
    <View>
      <ScreenHeader
        title={'Home'}
        image={require('~assets/images/screenHeaders/home.png')}
        imageWidth={184}
        imageHeight={75}
        imageFlushToBottom={true}
      />
      <ContentWrapper>
        <CardNavigationContainer />
      </ContentWrapper>
    </View>
  );
};

export default Home;
