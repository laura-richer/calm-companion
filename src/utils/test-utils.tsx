import React, { PropsWithChildren } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, RenderOptions } from '@testing-library/react-native';
import { Provider as StoreProvider } from 'react-redux';
import { EnhancedStore, configureStore } from '@reduxjs/toolkit';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { RootState } from '~store/index';
import navigationReducer, {
  initialState as navigationInitialState,
} from '~store/slices/navigation';
import quickHelpReducer, { initialState as quickHelpInitialState } from '~store/slices/quickHelp';

type DeepPartial<T> = {
  [K in keyof T]?: T[K] extends object ? DeepPartial<T[K]> : T[K];
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: EnhancedStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { preloadedState = {}, store = createTestStore(preloadedState) }: ExtendedRenderOptions = {}
) => {
  const wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </StoreProvider>
    );
  };

  return {
    ...render(ui, {
      wrapper,
    }),
  };
};

const _createStore = (preloadedState: Partial<RootState> = {}): EnhancedStore => {
  return configureStore({
    reducer: {
      navigation: navigationReducer,
      quickHelp: quickHelpReducer,
    },
    preloadedState,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        immutableCheck: { warnAfter: 256 },
        serializableCheck: { warnAfter: 256 },
      }),
  });
};

export const createTestStore = (preloadedState: DeepPartial<RootState> = {}): EnhancedStore => {
  const state: RootState = {
    // @ts-ignore
    navigation: {
      ...navigationInitialState,
      ...(preloadedState.navigation || {}),
    },
    // @ts-ignore
    quickHelp: {
      ...quickHelpInitialState,
      ...(preloadedState.quickHelp || {}),
    },
  };
  const store: EnhancedStore = _createStore(state);
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
};
