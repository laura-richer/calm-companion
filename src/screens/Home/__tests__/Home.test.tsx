import React from 'react';
import { screen } from '@testing-library/react-native';
import { renderWithProviders } from '~utils/test-utils';
import Home from '../Home';

const _renderDefault = () => {
  return renderWithProviders(<Home />);
};

describe('Home screen', () => {
  beforeEach(() => {
    _renderDefault();
  });

  test('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('renders screen header', () => {
    const screenHeader = screen.getByText('Home');

    expect(screenHeader).toBeOnTheScreen();
  });

  test('renders card navigation items', () => {
    const cardNavigationItems = screen.queryByLabelText('Navigation items');

    expect(cardNavigationItems).toBeOnTheScreen();
  });

  // test('does not render tab navigation', () => {
  //   const tabNavigation = screen.queryByLabelText('Tab navigation items');

  //   expect(tabNavigation).not.toBeOnTheScreen();
  // });

  // test('renders stack navigation', () => {
  //   const stackNavigation = screen.queryByLabelText('Stack navigation items');

  //   expect(stackNavigation).toBeOnTheScreen();
  // });
});
