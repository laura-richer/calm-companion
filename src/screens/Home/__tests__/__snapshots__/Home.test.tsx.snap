// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen matches snapshot 1`] = `
<View>
  <BVLinearGradient
    angle={170}
    angleCenter={
      {
        "x": 0.75,
        "y": 0.75,
      }
    }
    colors={
      [
        4282350466,
        4287349689,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    useAngle={true}
  >
    <View
      style={
        [
          {
            "padding": 10,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
      testID="screenHeaderContainer"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Poppins-Bold",
              "fontSize": 34,
            },
            {
              "color": "#ffffff",
              "marginBottom": 10,
            },
          ]
        }
      >
        Home
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
          }
        }
      >
        <Image
          accessibilityLabel="Home image"
          height={75}
          resizeMode="contain"
          source={
            {
              "testUri": "../../../src/assets/images/screenHeaders/home.png",
            }
          }
          style={
            [
              {
                "opacity": 0.1,
              },
              {
                "marginTop": 10,
              },
            ]
          }
          width={184}
        />
      </View>
    </View>
  </BVLinearGradient>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "paddingBottom": 210,
            "paddingHorizontal": 10,
            "paddingVertical": 30,
          }
        }
      >
        <View
          accessibilityLabel="Navigation items"
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              [
                {
                  "shadowColor": "#8b8490",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 8,
                },
                {
                  "flexBasis": "47%",
                  "marginBottom": 20,
                },
              ]
            }
          >
            <View
              accessibilityLabel="Quick Help"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "borderRadius": 8,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 14,
                    },
                    {
                      "color": "#3f7b82",
                      "maxWidth": 100,
                    },
                  ]
                }
              >
                Quick
                

                Help
              </Text>
              <Image
                accessibilityHint="Icon"
                alt="Quick Help icon"
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/images/screenHeaders/quick-help.png",
                  }
                }
                style={
                  {
                    "marginLeft": "auto",
                    "maxWidth": 75,
                    "opacity": 0.25,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "shadowColor": "#8b8490",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 8,
                },
                {
                  "flexBasis": "47%",
                  "marginBottom": 20,
                },
              ]
            }
          >
            <View
              accessibilityLabel="Anxiety Resources"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "borderRadius": 8,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 14,
                    },
                    {
                      "color": "#3f7b82",
                      "maxWidth": 100,
                    },
                  ]
                }
              >
                Anxiety
                

                Resources
              </Text>
              <Image
                accessibilityHint="Icon"
                alt="Anxiety Resources icon"
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/images/screenHeaders/anxiety-resources.png",
                  }
                }
                style={
                  {
                    "marginLeft": "auto",
                    "maxWidth": 75,
                    "opacity": 0.25,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "shadowColor": "#8b8490",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 8,
                },
                {
                  "flexBasis": "47%",
                  "marginBottom": 20,
                },
              ]
            }
          >
            <View
              accessibilityLabel="CBT Tools"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "borderRadius": 8,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 14,
                    },
                    {
                      "color": "#3f7b82",
                      "maxWidth": 100,
                    },
                  ]
                }
              >
                CBT
                

                Tools
              </Text>
              <Image
                accessibilityHint="Icon"
                alt="CBT Tools icon"
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/images/screenHeaders/cbt-tools.png",
                  }
                }
                style={
                  {
                    "marginLeft": "auto",
                    "maxWidth": 75,
                    "opacity": 0.25,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "shadowColor": "#8b8490",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 8,
                },
                {
                  "flexBasis": "47%",
                  "marginBottom": 20,
                },
              ]
            }
          >
            <View
              accessibilityLabel="Helpful Contacts"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "borderRadius": 8,
                    "padding": 10,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 14,
                    },
                    {
                      "color": "#3f7b82",
                      "maxWidth": 100,
                    },
                  ]
                }
              >
                Helpful
                

                Contacts
              </Text>
              <Image
                accessibilityHint="Icon"
                alt="Helpful Contacts icon"
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/images/screenHeaders/helpful-contacts.png",
                  }
                }
                style={
                  {
                    "marginLeft": "auto",
                    "maxWidth": 75,
                    "opacity": 0.25,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
